docs/java/how-to-set-activityoptions-in-java.md

```diff
@@ -0,0 +1,317 @@
+
+### `StartToCloseTimeout`
+
+- Type: `Duration`
+- Default: Defaults to [`ScheduleToCloseTimeout`](#scheduletoclosetimeout) value
+
+- With `ActivityStub`
+
+  ```java
+  GreetingActivities activities = Workflow.newActivityStub(GreetingActivities.class,
+              ActivityOptions.newBuilder()
+                      .setStartToCloseTimeout(Duration.ofSeconds(2))
+                      .build());
+  ```
+
+- With `WorkflowImplementationOptions`
+
+  ```java
+  WorkflowImplementationOptions options =
+              WorkflowImplementationOptions.newBuilder()
+                      .setActivityOptions(
+                              ImmutableMap.of(
+                                "EmailCustomerGreeting",
+                                      ActivityOptions.newBuilder()
+                                            // Set Activity Execution timeout (single run)
+                                            .setStartToCloseTimeout(Duration.ofSeconds(2))
+                                            .build()))
+                      .build();
+  ```
+
+See [Start-to-Close Timeout](/docs/concepts/what-is-a-start-to-close-timeout)
+
+### `HeartbeatTimeout`
+
+- Type: `Duration`
+- Default: None
+
+- With `ActivityStub`
+
+  ```java
+  private final GreetingActivities activities =
+      Workflow.newActivityStub(
+          GreetingActivities.class,
+          ActivityOptions.newBuilder()
+              .setHeartbeatTimeout(Duration.ofSeconds(2))
+              .build());
+  ```
+
+- With `WorkflowImplementationOptions`
+
+  ```java
+  WorkflowImplementationOptions options =
+              WorkflowImplementationOptions.newBuilder()
+                      .setActivityOptions(
+                              ImmutableMap.of(
+                                "EmailCustomerGreeting",
+                                      ActivityOptions.newBuilder()
+                                            .setHeartbeatTimeout(Duration.ofSeconds(2))
+                                            .build()))
+                      .build();
+  ```
+
+See [Heartbeat Timeout](/docs/concepts/what-is-a-heartbeat-timeout)
+
+### `TaskQueue`
+
+- Type: `String`
+- Default: Defaults to the Task Queue that the Workflow was started with.
+
+- With `ActivityStub`
+
+  ```java
+  GreetingActivities activities = Workflow.newActivityStub(GreetingActivities.class,
+                  ActivityOptions.newBuilder()
+                          .setTaskQueue("yourTaskQueue")
+                          .build());
+  ```
+
+- With `WorkflowImplementationOptions`
+
+  ```java
+  WorkflowImplementationOptions options =
+              WorkflowImplementationOptions.newBuilder()
+                      .setActivityOptions(
+                              ImmutableMap.of(
+                                "EmailCustomerGreeting",
+                                      ActivityOptions.newBuilder()
+                                            .setTaskQueue("yourTaskQueue")
```

Comment:
Same thing here, i think showing how to set these individually is awesome, just worrying that users will try to cut/paste and then get error that one of the required timeouts is missing.

Reply:
You're right; it makes sense to call this out. I'm going to add this info in the Activity heartbeat topic as well, but updating all samples with heartbeat to include starttoclosetimeout

Reply:
updated.